subprojects/core/src/main/java/org/gradle/internal/build/DefaultBuildLifecycleControllerFactory.java

```diff
@@ -60,6 +64,10 @@ public BuildLifecycleController newInstance(BuildDefinition buildDefinition, Ser
         }
 
         GradleInternal gradle = buildScopeServices.get(GradleInternal.class);
+        if (gradle.isRootBuild()) {
```

Comment:
❌ We should move the relevant TODO here like @mlopatkin suggested.

Reply:
I moved the attachment to the same place it was before, and kept the TODO that was there already